### ISSUE TITLE
cluster api controllers are not deleted after k8s master provision finished

### ISSUE THREAD

**Issue by gyliu513** [kubernetes-sigs/cluster-api#876]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**

I was using existing k8s cluster as bootstrap cluster, and after the k8s master was provisioned on OpenStack, I found the controllers was still running in my existing k8s cluster, I think those controllers, CRDs should be deleted on bootstrap cluster.

This is the command that I used to provision my cluster:
```shell
./clusterctl create cluster --bootstrap-cluster-kubeconfig /root/.kube/config --provider openstack  -c examples/openstack/out/cluster.yaml -m examples/openstack/out/machines.yaml  -p examples/openstack/out/provider-components.yaml
```
/cc @vincepri @detiber 

FYI @jichenjc @xunpan @clyang82 @morvencao

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by detiber** [comment]
@gyliu513 It sounds like you might be using an older version of the `clusterctl` source in the build that you are testing with (prior to the [pivot phase work](https://github.com/kubernetes-sigs/cluster-api/pull/731) being merged). Prior to that PR landing resources and provider components were indeed left stranded on the external bootstrap cluster, but afterwards they should be cleaned up regardless of the bootstrapping type: https://github.com/kubernetes-sigs/cluster-api/blob/master/cmd/clusterctl/phases/pivot.go

**Comment by gyliu513** [comment]
Thanks @detiber , let me test with the 0.1.0 cluster-api to see if it works.

**Comment by vincepri** [comment]
@gyliu513 Any updates on the issue above?

**Comment by gyliu513** [comment]
Thanks @vincepri for the reminder, yes, it was fixed in 0.1.0, please refer to our test result at https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/334#issuecomment-489428316
